### PR TITLE
update qml.device docs

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
Found in a previous PR. This docstring is quite stale, so I've updated the device list along with a few other things.

**Description of the Change:**
- Remove the interface-specific devices from the list (just use default.qubit!)
- Add lightning.qubit, default.qutrit, default.mixed, and a suggestion to use `qml.about()` to check what you have
- mention that some devices (eg default.qubit) may not need wires
- change the shots example to use a shot vector to suggest that it's possible/advertise the feature

**Benefits:**
people know about all the features available to them!

**Possible Drawbacks:**
still keeping this up-to-date? maybe we should just remove the explicit list of devices? I don't think it's too bad.

### CODE REVIEW

File: pennylane/__init__.py [hunk 1]
```diff
@@ -176,36 +177,39 @@ def device(name, *args, **kwargs):
...
 
-    * :mod:`'default.gaussian' <pennylane.devices.default_gaussian>`: a simple simulator
-      of Gaussian states and operations on continuous-variable circuit architectures.
+    * :mod:`'default.mixed' <pennylane.devices.default_mixed>`: a mixed-state
+      simulator of qubit-based quantum circuit architectures.
 
-    * :mod:`'default.qubit.tf' <pennylane.devices.default_qubit_tf>`: a state simulator
-      of qubit-based quantum circuit architectures written in TensorFlow, which allows
-      automatic differentiation through the simulation.
+    * ``'lightning.qubit'``: a more performant state simulator of qubit-based
```
Comment (albi3ro) [inline pennylane/__init__.py:183]: Worth including `lightning.kokkos` and `lightning.gpu` here too?

**Review by lillian542 — APPROVED** [review]
Looks good to me 🎉

File: pennylane/__init__.py [hunk 2]
```diff
@@ -176,36 +177,39 @@ def device(name, *args, **kwargs):
...
 
     Args:
         name (str): the name of the device to load
         wires (int): the number of wires (subsystems) to initialise
-            the device with
+            the device with. Note that this is optional for certain
+            devices, such as ``default.qubit``
 
     Keyword Args:
         config (pennylane.Configuration): a PennyLane configuration object
```
Comment (albi3ro) [inline pennylane/__init__.py:204]: Any idea what devices take a `Configuration`? I don't think I've ever actually touched this class.

Comment (timmysilv) [reply pennylane/__init__.py:204]: no clue, I have never touched it either, but I think it's still technically true.

**Review by albi3ro — APPROVED** [review]
We could potentially perform a deeper dive at some point, but definitely an improvement.

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**